### PR TITLE
Implements the lod view

### PR DESCRIPTION
**Pull request by wmelder**
related to this issue: https://github.com/beeldengeluid/beng-lod-server/issues/188

### CODE REVIEW

**Comment by wmelder** [comment]
https://github.com/beeldengeluid/beng-lod-server/issues/188

**Comment by wmelder** [comment]
@jblom Please have a look and run it locally, for example put in your browser: >http://127.0.0.1:5309/id/program/2101608190180567031

This is really the first version and not a minimal viable product yet. But maybe you can already check whether unittests are needed. Also, I could use some advice on styling, etc.

**Review by frankstrater — COMMENTED** [review]
Reviewed on request from @jblom

Looked at all the frontend code, skipped the Python part. General feedback:

- Use the latest version of Bootstrap 5.1.3
- Don't use inline styles, use Bootstrap classes
- Don't use MuseoSans, it's a commercial font and it's not licensed for this project. Use Raleway instead.
- Resize images (B&G logos) before including
- Remove files that are not used (either included bootstrap files or linked files)

Too many changes for a change request, so I suggest I'll do them in this branch when @wmelder is done with functional development.

**Comment by frankstrater** [comment]
Simple trick to resize images when you have ImageMagick installed:

`convert 'B&G_logo_RGB_liggend_RL.png' -resize 142 nisv_logo.png`

**Comment by gb-beng** [comment]
@wmelder perhaps we can use a vector image for the logo, such that it scales nicely? The huisstijl repository contains an svg as well. Just give me a thumbs up if you'd like me to make that change.

**Comment by frankstrater** [comment]
> @wmelder perhaps we can use a vector image for the logo, such that it scales nicely? Both the new and old png looks rather blurry on my (high-dpi) display. The huisstijl repository contains an svg as well. Just give me a thumbs up if you'd like me to make that change.

Updated it to SVG. New rule of thumb: use SVG unless there are compelling reasons not to (like supporting old browsers).

**Comment by wmelder** [comment]
wow. @jblom @frankstrater Thanks a lot for cleaning all this s*&!t  up! It looks all nice and clean API-wise and frontend-wise. I hope I will not ruin that too much with the upcoming changes I need to add ;-)

File: src/static/css/beng-lod-style.css [hunk 1]
```diff
@@ -15,10 +19,13 @@ a:hover {
 
 header {
 	background-color: #0f238c;
+    min-height: 210px;
 }
 
 header .row {
 	min-height: 150px;
+	font-size: 16px;
```
Comment (frankstrater) [inline src/static/css/beng-lod-style.css:27]: Deze is overbodig. Font-sizing nooit meer in px doen, rem only.

File: src/static/css/beng-lod-style.css [hunk 2]
```diff
@@ -15,10 +19,13 @@ a:hover {
...
+    font-size: 1.125rem
```
Comment (frankstrater) [inline src/static/css/beng-lod-style.css:28]: Altijd afsluiten met `;`

**Review by frankstrater — COMMENTED** [review]
Zie mijn comments in de CSS

File: src/static/css/beng-lod-style.css (hunk 2 above)
Comment (wmelder) [reply src/static/css/beng-lod-style.css:28]: oeps slordig..

File: src/static/css/beng-lod-style.css (hunk 1 above)
Comment (wmelder) [reply src/static/css/beng-lod-style.css:27]: thanks, font-size nooit meer in px => mental note

**Comment by wmelder** [comment]
@jblom after push only a unittest now fails. Can you please checkout and have a look?

**Comment by wmelder** [comment]
@jblom De backend functies zijn verbeterd, maar daadoor werken sommige tests niet meer. Kun jij uitchecken en wat programma's testen? Misschien ook de unittests aanpassen. Ook gtaa concepten kun je nu bekijken. 
NB voeg deze regel toe aan je hosts.conf: 
```
# reroute the domain for dev purposes
	127.0.0.1       data.beeldengeluid.nl
```
Daarna kun je gewoon de URI's doorklikken vanuit de browser.

**Comment by wmelder** [comment]
PS @frankstrater Styling is nagenoeg hetzelfde gebleven (en dus hopelijk nog steeds conform de richtlijnen).

**Review by frankstrater — APPROVED** [review]

File: src/templates/resource.html [hunk 3]
```diff
@@ -9,11 +9,24 @@
...
+				{% if (json_header|first) is defined %}
+					<h1>{{ ((json_header|first).pref_label|first) }}</h1>
+				{% endif %}
 				<a class="link-light" title="&lt;{{resource_uri}}&gt;" href="{{resource_uri}}" target="_blank">&lt;{{resource_uri}}&gt;</a>
 			</div>
 			<div class="col-6 text-end">
 				{% for item in json_header %}
-				<a class="link-light" title="&lt;{{item['o']}}&gt;" href="{{item['o']}}" target="_blank">&lt;{{item['namespace']}}<span class="fw-bold">{{item['property']}}</span>&gt;</a><br>
+					{% if item.o.prefix|length %}
+						<a title="&lt;{{item.o.uri}}&gt;"
```
[the remark this answers is not in the document]
Comment (frankstrater) [reply src/templates/resource.html:20]: I mean this for line 20, not 16...

File: src/static/css/beng-lod-style.css [hunk 4]
```diff
@@ -27,6 +27,14 @@ header .row {
     font-size: 1.125rem;
 }
 
+#header_rdf_type {
+    color: white
```
[the remark this answers is not in the document]
Comment (wmelder) [reply src/static/css/beng-lod-style.css:31]: aiaiai.. that's again the semicolon mistake... sorry about that..

File: src/templates/resource.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (wmelder) [reply src/templates/resource.html:20]: Thanks for you patience with a not so experienced front-ender...

Sorry for asking but to me it seems that it is better readable if I continue the attributes on a new line. It also does seem to work or it is rendered at least.. But, why is it that it is not a good practice to spread elements over multiple lines? What is the risk if you continue on a new line? Some browsers crash on that? Or...?

[the remark this answers is not in the document]
Comment (frankstrater) [reply src/templates/resource.html]: Some screenreaders for the visually impaired seem to have trouble with this kind of output. Unless the Python template parser will output it correctly ofcourse (need to check this). Apart from that it is conceptually more readable if you keep the standard distinction between block level and inline elements (it's also benifcial if you use code folding in your editor).

**Pull request merged**